### DISCUSSION TITLE
Daily logs feature in spdlog

### DISCUSSION THREAD

**Discussion by RahulVPatil24** [gabime/spdlog#2737] [Q&A] [answered]
I m implimenting normal below functionality as below 
#include "spdlog/sinks/daily_file_sink.h"
void daily_example()
{
    // Create a daily logger - a new file is created every day at 2:30 am
    auto logger = spdlog::daily_logger_mt("daily_logger", "logs/daily.txt", 2, 30);
}
whenever i run this code i have getting an logfile as an output but it is not giving me an any error  as i am not running it at 2:30 am am i doing any mistake to understand this feature ?

**Comment by tt4g** [comment]
What was different from your expectations?

**Reply by RahulVPatil24** [reply]
I want to know why above implimented function is still generating a logfile if it's not executing at 2:30am. Instead of giving an error.
as the are saying in the comment as  //Create a daily logger - a new file is created every day at 2:30 am

**Reply by tt4g** [reply, marked as answer]
To be precise, daily sink rotates at the specified time.
And the log file is created inside the constructor, so it is created even if the log output has never been made.

See also #1083
